strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticationProviderTest.java

```diff
@@ -81,6 +98,81 @@ public void embeddedLdapServerRegistersExpectedAuthenticationProvider()
                         new SimpleGrantedAuthority(SystemRole.REPOSITORY_MANAGER.name()),
                         new SimpleGrantedAuthority("USER_ROLE")
                 );
+        assertThat(passwordEncoder.matches("password", ldapUserDetails.getPassword())).isEqualTo(true);
+    }
+
+    @Test
+    public void base64EncodedPasswordAfterAlgorithmShouldWork()
+    {
+        userDetailsMapper.setBase64EncodedPassword(true);
+        ldapUserDetailsService.setUserDetailsMapper(userDetailsMapper);
+        UserDetails ldapUser = ldapUserDetailsService.loadUserByUsername("base64encoded-issue-1840");
+        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
+        LdapUserDetails ldapUserDetails = (LdapUserDetailsImpl) ldapUser;
+
+        assertThat(ldapUserDetails.getDn()).isEqualTo("uid=base64encoded-issue-1840,ou=Users,dc=carlspring,dc=com");
+        assertThat(ldapUserDetails.getUsername()).isEqualTo("base64encoded-issue-1840");
+        //Decoded Passsword of {MD5}X03MO1qnZdYdgyfeuILPmQ==
+        assertThat(ldapUserDetails.getPassword()).isEqualTo("{MD5}5f4dcc3b5aa765d61d8327deb882cf99");
+        assertThat(passwordEncoder.matches("password", ldapUserDetails.getPassword())).isEqualTo(true);
+
+        assertThat(((List<SimpleGrantedAuthority>) ldapUser.getAuthorities()))
+                .contains(
+                        new SimpleGrantedAuthority(SystemRole.REPOSITORY_MANAGER.name()),
+                        new SimpleGrantedAuthority("USER_ROLE")
+                );
+
+        userDetailsMapper.setBase64EncodedPassword(false);
+    }
+
+    @Test
+    public void base64EncodedPasswordAfterAlgorithmShouldFail()
+    {
+        ldapUserDetailsService.setUserDetailsMapper(userDetailsMapper);
+        userDetailsMapper.setBase64EncodedPassword(false);
+        UserDetails ldapUser = ldapUserDetailsService.loadUserByUsername("base64encoded-issue-1840");
+        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
+        LdapUserDetails ldapUserDetails = (LdapUserDetailsImpl) ldapUser;
+        assertThat(passwordEncoder.matches("password", ldapUserDetails.getPassword())).isEqualTo(false);
+    }
+
+    @Test
+    public void base64EncodedPasswordWithAlgorithmShouldWork()
+    {
+        userDetailsMapper.setBase64EncodedPassword(true);
+        ldapUserDetailsService.setUserDetailsMapper(userDetailsMapper);
+        UserDetails ldapUser = ldapUserDetailsService.loadUserByUsername("base64withalgorithm-issue-1840");
+
+        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
```

Comment:
Please, always add an extra line before and after assertions (or assertion blocks). This makes it easier to read and review. Thanks! :)
```suggestion
        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
        
```